app/utils/validators/forms/FieldValidator.scala

```diff
@@ -151,8 +151,10 @@ object FieldValidator {
   private val emailPattern = Pattern.compile("""^\S+@\S+$""")
   val isValidEmail: String => Boolean = (name: String) => emailPattern.matcher(name).matches()
 
-  private val eoriPattern = Pattern.compile("^[a-zA-Z]{2}[\\da-zA-Z]+$")
-  val isValidEORIPattern: String => Boolean = (name: String) => eoriPattern.matcher(name).matches()
+  val isValidEori: String => Boolean = (eori: String) => {
+    val (countryCode, number) = eori.splitAt(2)
+    countryCode.forall(_.isLetter) && number.forall(_.isDigit) && isInRange(12, 15)(number.length)
```

Comment:
Are we sure about 12-15 characters?

I remember that we had something like this before. EORI number in Sweden has two country code characters + 10 digits

Reply:
Checking this back with the business.

Reply:
I remeber that there was something with adding zeros at the end, not sure.

In general PR looks good, if you confirm this with BAs you can merge it. I'll leave the approve now :)

Reply:
You're right @PatrykRudnicki. It was not taken into account that we need to accept foreign EORI numbers as well. Update-test loop. Again...

Reply:
okay :)